packages/codemod-utils/src/function.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+  ASTPath,
+  ArrowFunctionExpression,
+  Collection,
+  FunctionDeclaration,
+  FunctionExpression,
+  JSCodeshift,
+} from "jscodeshift";
+
+import { getDefaultExport, isDefinedInGlobalScope } from "./global.js";
+
+export type FunctionLike =
```

Comment:
Im still not sure if its a good idea to combine multiple node types into single type. From usability perspective it looks good, because we can have same utils for different kinds of functions (Declarations and ArrowFunctions).